### ISSUE TITLE
Contribution to JWOC 2024

### ISSUE THREAD

**Issue by anand-harsh** [anand-harsh/Edumi#31]
# Welcome to Edumi Open Source Project

Thank you for considering contributing to Edumi! We appreciate your interest in helping us improve our educational platform.

## How to Contribute

You can contribute to Edumi in several ways:

1. **Working on Existing Issues:**
   - Check out our [issues page](https://github.com/anand-harsh/Edumi/issues) to find tasks that need attention.
   - Comment on the issue you would like to work on to avoid duplication of efforts.
   - Fork the repository, make your changes, and submit a pull request.

2. **Requesting Features or Enhancements:**
   - If you have ideas for new features or improvements, please open an issue with the details.
   - Discuss your ideas with the community and the maintainers to refine the proposal.

3. **Reporting Bugs:**
   - If you find any bugs or issues, please open a detailed issue with information on how to reproduce it.
   - Be specific about your environment, steps to reproduce, and any error messages.

## Contributing Guidelines

Before contributing, please read our [Contributing Guidelines](https://github.com/anand-harsh/Edumi/blob/master/CONTRIBUTION.md) to understand the project's standards and processes.

## Code Comments

We encourage contributors to add comments to the code they submit. Clear and concise comments help everyone understand the purpose and functionality of the code.

## Getting Started

To get started with Edumi development, follow these steps:
1. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the repository to your GitHub account.
2. [Clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) the forked repository to your local machine.
3. Create a new branch for your changes: `git checkout -b feature-name`.
4. Make your changes and add comments to the code.
5. Commit your changes: `git commit -m "Your descriptive commit message"`.
6. Push the changes to your fork: `git push origin feature-name`.
7. [Create a pull request](https://docs.github.com/en/pull-requests/creating-and-managing-pull-requests/creating-a-pull-request) on the main repository.

Thank you for contributing to Edumi! We look forward to working with you. ✨✨

Happy coding!

**Comment by anand-harsh** [comment]
# Update on Contributing points

1) One user can work on only one issue at a time.
2) User can open new issue only after successful merging of previous issue.
3) Issue will be marked as Easy, Medium , Hard

# Days to complete a issue

Easy : 1 day
Medium : 2-3 day
Hard : 4days

# Points for Issue marking (Phase 1 )

Easy : 1 points
Medium: 3 points
Hard : 5 points

**Comment by anand-harsh** [comment]
# Backend Contributions

1) Get specific course using id, having multiple lectures in a Course.
Issue: #40 

2)  Add Lectures to specific course
Issue: 

3) Make authorisation middleware for admin role
Issue: 

4) Delete a whole course
Issue: 

5) Get all users (Admin Controller)
Issue: 

6) Update user roles for any user  (Admin Controller)
Issue: 

7) Delete any user  (Admin Controller)
Issue: 

8) Delete personal profile / account
Issue:

**Comment by anand-harsh** [comment]
# Update on Contributing points

1) One user can work on only one issue at a time.
2) User can open new issue only after successful merging of previous issue.
3) Issue will be marked as Easy, Medium , Hard

# Days to complete a issue

Easy : 1 day
Medium : 2-3 day
Hard : 4 days

# Points for Issue marking (Phase 2 )

Easy : 2 points
Medium: 4 points
Hard : 8 points

You can Contribute to " not-assigned " issues after assigning.
